### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by Leeoasis**
## Completed tasks in this PR

- [ ] Create a class Classroom.
- [ ] Create the relationship between Classroom and Student.
- [ ] Create a class Book.
- [ ] Create a class Rental.
- [ ] Create the relationship between Person and Book using the intermediate class Rental.

### CODE REVIEW

File: students.rb [hunk 1]
```diff
@@ -3,8 +3,8 @@
 class Student < Person
   attr_accessor :classroom
 
-  def initialize(age, classroom, name: 'Unknown', parent_permission: true)
-    super(age, name, parent_permission)
+  def initialize(classroom, age, name: 'Unknown', parent_permission: true)
+    super(age, name: name, parent_permission: parent_permission)
     @classroom = classroom
   end
```
Comment (julie-ify) [inline students.rb]: - [ ] The Student class should have a method of associating with the Classroom class.

  ```
  def classroom=(classroom)
    @classroom = classroom
    classroom.students.push(self) unless classroom.students.include?(self)
  end
  ```

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(date, person)
+    rental = Rental.new(date, self, person)
+    @rentals << rental
+    rental
+  end
```
Comment (julie-ify) [inline book.rb]: - [ ] There is no need to add to the `rentals` object again. Creating a rental object does that already. Kindly remove the last two lines and `rental = `

  ```
  def add_rental(date, person)
    Rental.new(date, self, person)
  end
  ```

File: person.rb [hunk 3]
```diff
@@ -1,18 +1,26 @@
...
 
   def can_use_services?
     of_age? || @parent_permission
   end
 
+  def add_rental(date, book)
+    rental = Rental.new(date, book, self)
+    @rentals << rental
+    rental
+  end
```
Comment (julie-ify) [inline person.rb]: - [ ] Kindly only leave the rental object creation line.

  ```
  def add_rental(date, book)
    Rental.new(date, book, self)
  end
  ```

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @Leeoasis,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Lizdev-05 — APPROVED** [review]
Hi @Leeoasis, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

- Good commit history 💰

- Good PR title/description 💯

- All linters passed 🏅

- Personalised README file 👍

- Good code organization and working app ✔️


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**